packages/loopring_v3/circuit/Circuits/AccountNewCircuit.h

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2017 Loopring Technology Limited.
+#ifndef _NEWACCOUNTCIRCUIT_H_
+#define _NEWACCOUNTCIRCUIT_H_
+
+#include "Circuit.h"
+#include "../Utils/Constants.h"
+#include "../Utils/Data.h"
+
+#include "ethsnarks.hpp"
+#include "utils.hpp"
+
+using namespace ethsnarks;
+
+namespace Loopring
+{
+
+class AccountNewCircuit : public BaseTransactionCircuit
+{
+public:
+
+    // Inputs
+    DualVariableGadget payerAccountID;
+    DualVariableGadget feeTokenID;
+    DualVariableGadget fee;
+    DualVariableGadget nonce;
+    DualVariableGadget newAccountID;
+    DualVariableGadget newOwner;
+    VariableT newPublicKeyX;
+    VariableT newPublicKeyY;
+    DualVariableGadget walletHash;
+
+    // Signature
+    Poseidon_gadget_T<11, 1, 6, 53, 10, 1> hash;
+
+    // Validate
+    EqualGadget isNewAccountTx;
+    RequireNotZeroGadget requireNewOwnerNotZero;
+    IfThenRequireEqualGadget requireAccountLeafEmpty;
+
+    // Compress the public key
+    CompressPublicKey compressPublicKey;
+
+    // Balances
+    DynamicBalanceGadget balanceS_A;
+    DynamicBalanceGadget balanceB_O;
+    // Fee as float
+    FloatGadget fFee;
+    RequireAccuracyGadget requireAccuracyFee;
+    // Fee payment from From to the operator
+    TransferGadget feePayment;
+
+    // Increase the nonce
+    AddGadget nonce_after;
```

Comment:
In AddGadget, if we use SNARK_SCALAR_FIELD=11, if A=6 and B=8, then `unsafeAdd(A, B).result()` will be 4. 4 will pass the rang check of 4 bytes (11 only require 4 bytes). I think only the rangcheck will not work.

Reply:
As discussed in call, inputs need to within the expected range, otherwise the gadgets will indeed not work. This is why most of the times the gadgets take in the expected max bit size of the inputs to make sure they will work correctly, but this is purely a way to find potential issues and does not have any effect on the circuit itself.